pylint/checkers/refactoring/refactoring_checker.py

```diff
@@ -684,18 +708,42 @@ def _check_consider_using_comprehension_constructor(self, node):
                 message_name = "consider-using-set-comprehension"
                 self.add_message(message_name, node=node)
 
+    def _check_consider_using_generator(self, node):
+        # We only check 'any' and 'all' because for list, set, and tuple a generator performs worse
+        # See https://github.com/PyCQA/pylint/pull/3309#discussion_r576683109
+        checked_call = ["any", "all"]
+        if (
+            isinstance(node, astroid.Call)
+            and node.func
+            and isinstance(node.func, astroid.Name)
+            and node.func.name in checked_call
+        ):
+            # any or all calls takes exactly one argument
+            if len(node.args) == 1 and isinstance(node.args[0], astroid.ListComp):
+                inside_comp = node.args[0].as_string()[
+                    1:-1
+                ]  # remove square brackets '[]'
```

Comment:
I assume this is safe with regard to leading/trailing space(s) (inside the square brackets)?

Reply:
Yes it is.
![image](https://user-images.githubusercontent.com/20282025/108061084-622f5900-708a-11eb-90b9-c5267a47769b.png)